### DISCUSSION TITLE
Set compiler in general (\setPlaceholder*{__ROBEXT_LATEX_ENGINE__})

### DISCUSSION THREAD

**Discussion by dflvunoooooo** [leo-colisson/robust-externalize#18] [Q&A] [answered]
More like a question than an issue. 

Is `\setPlaceholder*{__ROBEXT_LATEX_ENGINE__}{xelatex}` the desired way to set the compiler in general for every environment?

**Comment by tobiasBora** [comment, marked as answer]
While it works, there are a number of way to make it cleaner. I would rather recommend you to put that in a preset in order to make changes as local as possible (maybe another package assumes that `__ROBEXT_LATEX_ENGINE__` is equal to pdflatex by default and it is important for them?). In the `latex`-based presets, you can even use `use xelatex` instead of `set placeholder={__foo__}{bar}`. So I would rather recommend:
```
\robExtConfigure{
  add to preset={latex}{
    use xelatex,
  },
}
```
but it is certainly a matter of taste.

Note also that the star version of `\setPlaceholder` is good here (because this placeholder is automatically imported later), but it does not automatically import the placeholder (I needed to do that for efficiency reasons… I might have some ideas to avoid the need to import placeholders but I need to test it before), so for user-defined placeholders, do not put any `*` if you do not want to worry about the import system.

**Comment by tobiasBora** [comment]
Also, you might prefer the discussions forum https://github.com/leo-colisson/robust-externalize/discussions for questions (but do add stuff here if it is a bug or a feature request). The advantage of the forum is that others will see my answers, while here it will be hidden once the issue is closed. If you don't mind I'll convert it to a discussion now.

**Comment by dflvunoooooo** [comment]
> While it works, there are a number of way to make it cleaner. I would rather recommend you to put that in a preset in order to make changes as local as possible (maybe another package assumes that `__ROBEXT_LATEX_ENGINE__` is equal to pdflatex by default and it is important for them?). In the `latex`-based presets, you can even use `use xelatex` instead of `set placeholder={__foo__}{bar}`. So I would rather recommend:
> 
> ```
> \robExtConfigure{
>   add to preset={latex}{
>     use xelatex,
>   },
> }
> ```
> 
> but it is certainly a matter of taste. …

Yes, I would prefer it that way too. And since the latex preset is applied to all environments, it achieves the same.
